File: vaxsul_backend/src/main/kotlin/com/github/dannful/domain/model/Vaccine.kt
package com.github.dannful.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Vaccine(
    val pricePerUnit: Float,
    val amountInStock: Int,
    val researchId: Int,
    val dose: Int,
    val description: String,
    val name: String
)

@Serializable
data class IdVaccine(
    val id: Int,
    val pricePerUnit: Float,
    val amountInStock: Int,
    val researchId: Int,
    val dose: Int,
    val description: String,
    val name: String
)

File: vaxsul_frontend/src/types/vaccine.ts
import { z } from "zod";

export const VaccineSchema = z.object({
  id: z.number(),
  dose: z.number(),
  pricePerUnit: z.number(),
  amountInStock: z.number(),
  researchId: z.number().optional().nullable(),
  sellable: z.boolean(),
  name: z.string(),
  laboratory: z.string(),
  description: z.string(),
});

export const VaccineSearchSchema = z.object({
  minimumPrice: z.number().optional(),
  maximumPrice: z.number().optional(),
  name: z.string().optional(),
  id: z.number().optional(),
});

export type Vaccine = z.infer<typeof VaccineSchema>;
export type VaccineSearch = z.infer<typeof VaccineSearchSchema>;
